Plugin Management: porting new plugin list changes to Calypso Blue

yashwin opened the pull request:
#### Proposed Changes

This PR makes necessary changes to the plugins' view, making it available to Calypso Blue.

#### Testing Instructions

**Prerequisites**

Since this change is made specifically for agencies, you must set yourself(partner) as an agency to test the changes in Calypso Green - 2c49b-pb. Make sure to switch it back to the previous type. 

**Instructions**

1. Run `git checkout add/porting-new-plugin-management-ui-to-calypso-blue` and `yarn start`
2. After it is done, run `yarn start-jetpack-cloud-p`.
3. Open http://jetpack.cloud.localhost:3001/, and you'll be redirected to the /dashboard.
4. Click on **Plugins** on the sidebar and verify everything looks and works as expected.
5. Open http://calypso.localhost:3000/plugins/manage and verify the new plugin UI works as expected.
6. Also verify the changes are looking good in http://calypso.localhost:3000/plugins/manage/:site.

All site view

<img width="1644" alt="Screenshot 2022-09-19 at 12 06 21 PM" src="https://user-images.githubusercontent.com/10586875/190966930-07382ead-f739-46ca-87cc-a65573383925.png">

<img width="1646" alt="Screenshot 2022-09-19 at 12 35 26 PM" src="https://user-images.githubusercontent.com/10586875/190967109-bbc10b12-569d-4369-be74-1eb432d6e598.png">

<img width="410" alt="Screenshot 2022-09-19 at 12 06 31 PM" src="https://user-images.githubusercontent.com/10586875/190967093-9e7c20a0-4832-432f-b7c1-79365e2daa34.png">

Single site view

<img width="801" alt="Screenshot 2022-09-20 at 12 04 26 PM" src="https://user-images.githubusercontent.com/10586875/191184627-cd8c68dc-27ac-43d3-935f-f5441037615b.png">
<img width="359" alt="Screenshot 2022-09-20 at 12 04 45 PM" src="https://user-images.githubusercontent.com/10586875/191184647-3756de2e-2fad-40d4-a22d-8610421671bc.png">
<img width="1570" alt="Screenshot 2022-09-20 at 12 05 22 PM" src="https://user-images.githubusercontent.com/10586875/191184650-af8130b4-ef34-4540-9fa0-57ceb2f0f3a3.png">


> **_NOTE:_** Other changes like moving the actions to the header will be made in a different PR.

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into the trunk. Inapplicable items can be left unchecked.

The PR author and reviewer are responsible for completing the checklist.

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes? Will be added later - 1202518759611394-as-1202627702018877
- [x] ~~Have you tested the feature in [Simple](P9HQHe-k8-p2-p2), [Atomic](P9HQHe-jW-p2-p2), and [self-hosted Jetpack sites](PCYsg-g6b-p2-p2)?~~
- [x] Have you checked for TypeScript, React, or other console errors?
- [x] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [x] Have we sent any new strings [for translation](PCYsg-1vr-p2-p2) ASAP?

Related to 1202518759611394-as-1202698250096217 & 1202518759611394-as-1202698250096219

yashwin committed:
port new plugin list changes to calyso blue

yashwin committed:
Update title

yashwin committed:
Minor css refactor and style changes

yashwin committed:
Remove Bulk Action test cases as that code is removed

yashwin committed:
Minor css refactor

yashwin committed:
Fix remove plugin bulk action

yashwin committed:
Remove unnecessary code

yashwin commented on client/my-sites/plugins/plugins-list/test/fixtures/index.js:
```diff
@@ -1,58 +0,0 @@
-export const sites = [
```
Removed the test cases here since these are not a part of this component.

grbicsanja commented:
I noticed a discrepancy with the toggle component color. This is what I see in the PR:
<img width="1440" alt="Screen Shot 2022-09-19 at 12 10 12" src="https://user-images.githubusercontent.com/90194580/190996829-64691320-d6f3-4f80-99a5-c973ba4e2deb.png">
And in production, this is the toggle component used all across the board. We should stick to using the same one on WPCOM (green active state):
<img width="1440" alt="Screen Shot 2022-09-19 at 12 10 17" src="https://user-images.githubusercontent.com/90194580/190996959-3235491b-631d-447c-aa06-28008bf93230.png">
<img width="1440" alt="Screen Shot 2022-09-19 at 12 10 00" src="https://user-images.githubusercontent.com/90194580/190996999-660a4a6b-e6fc-4233-b989-f025a426c28d.png">


Other than that, LGTM!

yashwin committed:
Set toggle active color to green

yashwin commented:
> I noticed a discrepancy with the toggle component color. 

Thanks for the review, @grbicsanja. The changes have been made.

vitozev commented on client/my-sites/plugins/main.jsx:
```diff
@@ -488,86 +488,59 @@ export class PluginsMain extends Component {
 							<div className="plugins__main-header">
 								<SectionNav
-									applyUpdatedStyles={ isJetpackCloud }
+									applyUpdatedStyles
```
Can we apply the custom CSS (`.section-nav-updated {}`) required for the plugin management UI outside this component? The `applyUpdatedStyles` parameter is too generic, but in reality (correct me if I'm wrong), these styles make sense only in the context of plugin management UI, so IMO these styles should be moved outside of the `<SectionNav />`, and this parameter should be removed at all.

If someone tomorrow decides to use the `<SectionNav>` component, the `applyUpdatedStyles` might be used incorrectly because it mentions updated styles.

yashwin replied:
We are now using this new style in Agency Dashboard and Plugin Management. But, the plan is to use this throughout Jetpack Cloud.

vitozev replied:
Got it! It is still a bit confusing because `<SectionNav />` is used quite a lot (I think) in WordPress.com.

Maybe we can document this in the README file of the SectionNav component in this case?

yashwin replied:
Done - https://github.com/Automattic/wp-calypso/pull/67910/commits/059f630ec4da1ae8ed87a7f15bee6f75f490a401

vitozev commented on client/my-sites/plugins/main.jsx:
```diff
@@ -488,86 +488,59 @@ export class PluginsMain extends Component {
+								hideClose={ ! this.props.search }
+								ref={ `url-search` }
+								analyticsGroup="Plugins"
+								placeholder={ this.props.translate( 'Search plugins' ) }
```
Can we use the `getSearchPlaceholder()` function here? If not, we should probably remove it as it's not used anymore.

yashwin replied:
I'll remove the `getSearchPlaceholder` function

vitozev commented on client/my-sites/plugins/main.jsx:
```diff
@@ -488,86 +488,59 @@ export class PluginsMain extends Component {
 									</NavTabs>
-									{ ! isJetpackCloud && (
-										<Search
-											pinned
-											fitsContainer
```
I'm not sure what these two properties are all about, but I'm curious if they can help us improve the UX? If not, you can ignore this comment.

yashwin replied:
No, these are not needed. This was needed for the old UI

yashwin committed:
Update active styles on single buttons

vitozev commented on client/my-sites/plugins/plugins-list/index.jsx:
```diff
@@ -609,34 +582,19 @@ export class PluginsList extends Component {
-				) : (
-					<>
-						<Card className={ itemListClasses }>
-							{ this.orderPluginsByUpdates( this.props.plugins ).map( this.renderPlugin ) }
```
Maybe that's a good candidate for a follow-up task? To order the plugins awaiting update on the top of the list. What do you think?

yashwin replied:
Yes. But, should we do that? I can add a task to make this change if you would like to.

vitozev replied:
I'm not sure about it. @grbicsanja, what do you think?

grbicsanja replied:
I like the idea, when there's many plugins, the ones that have updates available can easily stay out of site. I'd say we follow up with that task 👍

vitozev reviewed (commented):
I did manual testing on both Calypso Blue and Jetpack Cloud and all the features are working as expected (tested both mobile and desktop UI).

---

In Calypso Blue we still display the following feedback notices:

![image](https://user-images.githubusercontent.com/1749918/191204658-fde4c69b-5927-456b-ae83-2381182c8222.png)

We can hide them by removing the following piece of code in `plugins-list/index.jsx`:

```
{ ! this.props.isJetpackCloud && (
	<PluginNotices sites={ this.getPluginsSites() } plugins={ this.props.plugins } />
) }
```

Besides that I left some comments and thoughts which we can discuss.

yashwin committed:
Remove plugin notices and unused code

yashwin commented:
Thanks for the review, @vitozev. I made addressed your comments and made some changes - https://github.com/Automattic/wp-calypso/pull/67910/commits/5e52a62ee263f0318aa20b96ecd57799c8f2baf0

yashwin committed:
Update README.md for <SectionNav />

vitozev reviewed (approved):
LGTM! Nice job!

grbicsanja reviewed (approved):
LGTM! Nice work!! 🙌
